src/main/java/au/org/ala/biocache/dto/SpatialSearchRequestDTO.java

```diff
@@ -58,6 +60,12 @@ public static SpatialSearchRequestDTO create(SpatialSearchRequestParams params){
                 dto.setStart(0);
             }
         }
+
+        // Remove duplicate fq, remove duplicated q
+        if (dto.fq != null && dto.fq.length > 0) {
+            dto.fq = Arrays.stream(dto.fq).filter(Predicate.not(dto.q::equals)).distinct().toArray(String[]::new);
```

Comment:
comparing dto.fq with dto.q ? or it is a typo?

Reply:
fq can contain duplicates of the q.